### ISSUE TITLE
terraform fails if you change the list of vars on a template

### ISSUE THREAD

**Issue by ketzacoatl** [hashicorp/terraform#2583]
When I added a new variable to a template, Terraform incessantly errored out with:

```
$ tfp
Error configuring: 1 error(s) occurred:

* Resource 'template_file.minion-init' does not have attribute 'rendered' for variable 'template_file.minion-init.rendered'
```

Dropping the variable I added did not help. I probably could have tainted a few resources, but I eventually just rm'd everything and recreated.

What is the proper way to added/change the list of variables in for a template resource in an active environment?

**Comment by thegedge** [comment]
I'd be interested in knowing if this bug is specific to templates, or a more general error. I see it a ton of times where `terraform plan` says `Resource 'xyz' does not have attribute 'foo' for 'xyz.foo'`.

**Comment by sourcec0de** [comment]
It seems to be template specific. I am expereincing the same bug. I think doing the same thing as @ketzacoatl I am trying to build a launch config for kubernetes minions and inject the master private ip.

```
resource "aws_launch_configuration" "k8s_minion" {
    name = "k8s_minion"
    image_id = "${lookup(var.base_amis, var.aws_region)}"
    instance_type = "${var.k8s_minion_instance_size}"
    key_name = "${aws_key_pair.deployer.key_name}"
    enable_monitoring = true
    user_data = "${template_file.minion.rendered}"
    security_groups = [
        "${aws_security_group.minion.id}",
        "${aws_security_group.admin.id}"
    ]

    root_block_device {
        volume_type = "gp2"
        volume_size = 50
        delete_on_termination = true
    }
}

resource "template_file" "minion" {
    filename = "./templates/node.yml"

    vars {
        master_ip = "${aws_instance.k8s_master.private_ip}"
    }
}

output "node.yml" {
    value = "${template_file.minion.rendered}"
}
```

Renders this error
```
Resource 'template_file.minion' does not have attribute 'rendered' for variable 'template_file.minion.rendered'
```

**Comment by chiefy** [comment]
:+1: 
```
...
output "part-001.conf" {
    value = "${template_file.user-data-script.rendered}"
}
...
```

I get the same thing when trying any kind of `plan` command.

**Comment by thegedge** [comment]
@sourcec0de I should've been less vague on why I was asking that question :)

So to clarify, I've also seen something similar for `aws_security_group` where I'd get the error:

```
Resource 'aws_security_group.foo' does not have attribute 'id' for variable 'aws_security_group.foo.id'```

**Comment by timfallmk** [comment]
I'm getting something similar with #3362

**Comment by cleung2010** [comment]
I am also encountering this, any solutions/workarounds?

**Comment by cleung2010** [comment]
Nevermind, I was using count on `template_file` but not using an index when calling `rendered`.

**Comment by brikis98** [comment]
Seeing the same error as @thegedge with `aws_security_group`:

```
Resource 'aws_security_group.ecs' does not have attribute 'id' for variable 'aws_security_group.ecs.id'
```

**Comment by mitchellh** [comment]
This should be fixed with 0.8!
